central/declarativeconfig/updater/updater.go

```diff
@@ -0,0 +1,14 @@
+package updater
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/proto"
+)
+
+// ResourceUpdater handles updates of proto resources within declarative config reconciliation routine.
+// Each ResourceUpdater is responsible for updates of specific proto type.
+// TODO(ROX-14694): Extend interface with methods necessary for resource deletion.
+type ResourceUpdater interface {
```

Comment:
super-nit: I wonder whether just calling it `ResourceHandler` would be more clear, given that it will handle both create / delete, but 🤷 
I don't have too strong of an opinion to block on it, so feel free to ignore that one.

Reply:
I guess what prevented me from going for `ResourceHandler` is the fact that we already have a concept of handlers in declarative configuration, as well as a field in `manager_impl` called `transformedMessagesByHandler`